From 8c88bc1a64186ab993bca2410e6e1779ca59948f Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Thu, 30 Jun 2011 12:46:55 +0200
Subject: [PATCH 24/32] pbutils: fix compilation error in macro expansion

Fix:
 "encoding-target.c: In function 'get_locale': encoding-target.c:421: error:
 used struct type value where scalar is required"

Rename a local variable from 'loc' to 'locale' since GST_LOG declares
a variable called loc internally.
---
 gst-libs/gst/pbutils/encoding-target.c |   14 +++++++-------
 1 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/gst-libs/gst/pbutils/encoding-target.c b/gst-libs/gst/pbutils/encoding-target.c
index 29e67d0..79c340d 100644
--- a/gst-libs/gst/pbutils/encoding-target.c
+++ b/gst-libs/gst/pbutils/encoding-target.c
@@ -412,16 +412,16 @@ serialize_stream_profiles (GKeyFile * out, GstEncodingProfile * sprof,
 static gchar *
 get_locale (void)
 {
-  const char *loc = NULL;
+  const char *locale = NULL;
   gchar *ret;
 
 #ifdef ENABLE_NLS
 #if defined(LC_MESSAGES)
-  loc = setlocale (LC_MESSAGES, NULL);
-  GST_LOG ("LC_MESSAGES: %s", GST_STR_NULL (loc));
+  locale = setlocale (LC_MESSAGES, NULL);
+  GST_LOG ("LC_MESSAGES: %s", GST_STR_NULL (locale));
 #elif defined(LC_ALL)
-  loc = setlocale (LC_ALL, NULL);
-  GST_LOG ("LC_ALL: %s", GST_STR_NULL (loc));
+  locale = setlocale (LC_ALL, NULL);
+  GST_LOG ("LC_ALL: %s", GST_STR_NULL (locale));
 #else
   GST_LOG ("Neither LC_ALL nor LC_MESSAGES defined");
 #endif
@@ -429,11 +429,11 @@ get_locale (void)
   GST_LOG ("i18n disabled");
 #endif
 
-  if (loc == NULL || g_ascii_strncasecmp (loc, "en", 2) == 0)
+  if (locale == NULL || g_ascii_strncasecmp (locale, "en", 2) == 0)
     return NULL;
 
   /* en_GB.UTF-8 => en */
-  ret = g_ascii_strdown (loc, -1);
+  ret = g_ascii_strdown (locale, -1);
   ret = g_strcanon (ret, "abcdefghijklmnopqrstuvwxyz", '\0');
   GST_LOG ("using locale: %s", ret);
   return ret;
-- 
1.7.1

